starfish/pipeline/spots/gaussian.py

```diff
@@ -0,0 +1,124 @@
+        self.intensities = None
+
+    @classmethod
+    def from_cli_args(cls, args):
+        s = Stack()
+        s.read(args.input)
+        return cls(s)
+
+    @classmethod
+    def get_algorithm_name(cls):
+        return "gaussian_spot_detector"
+
+    @classmethod
+    def add_arguments(cls, group_parser):
+        # todo better description of how aux_image gets used
+        group_parser.add_argument("--blobs", type=str, help='aux image key')
+        group_parser.add_argument(
+            "--min-sigma", default=4, type=int, help="Minimum spot size (in standard deviation)")
+        group_parser.add_argument(
+            "--max-sigma", default=6, type=int, help="Maximum spot size (in standard deviation)")
+        group_parser.add_argument("--num-sigma", default=20, type=int, help="Number of scales to try")
+        group_parser.add_argument("--threshold", default=.01, type=float, help="Dots threshold")
+        group_parser.add_argument("--show", default=False, type=bool, help="Dots threshold")
+
+    def detect(
+            self, min_sigma, max_sigma, num_sigma, threshold, blobs, measurement_type='max', bit_map_flag=False,
+            *args, **kwargs):
+        self.blobs = self.stack.aux_dict[blobs]
+        fitted_blobs = self._fit(min_sigma, max_sigma, num_sigma, threshold)
+        spots_df_viz = self._fitted_blobs_to_df(fitted_blobs)
+        intensity = self._measure(self.blobs, spots_df_viz, measurement_type)
+        spots_df_viz['intensity'] = intensity
+        spots_df_viz['spot_id'] = spots_df_viz.index
+        self.spots_df_viz = spots_df_viz  # todo this needs a class if it's going to stay around
+        self.intensities = self._measure_stack(measurement_type)
+        res = self._to_encoder_dataframe(bit_map_flag)
+        return res
+
+    def _fit(self, min_sigma, max_sigma, num_sigma, threshold):
+        fitted_blobs = blob_log(self.blobs, min_sigma, max_sigma, num_sigma, threshold)
+        fitted_blobs[:, 2] = fitted_blobs[:, 2] * np.sqrt(2)
+        self.num_objs = fitted_blobs.shape[0]
+        return fitted_blobs
+
+    @staticmethod
+    def _fitted_blobs_to_df(fitted_blobs):
+        res = pd.DataFrame(fitted_blobs)
+        res['d'] = res[2] * 2
+        res.rename(columns={0: 'x', 1: 'y', 2: 'r'}, inplace=True)
+        res['xmin'] = np.floor(res.x - res.r)
+        res['xmax'] = np.ceil(res.x + res.r)
+        res['ymin'] = np.floor(res.y - res.r)
+        res['ymax'] = np.ceil(res.y + res.r)
+        return res
+
+    @staticmethod
+    def _measure(img, spots_df, measurement_type):
+        res = []
+        for row in spots_df.iterrows():
+            row = row[1]
+            subset = img[int(row.xmin):int(row.xmax), int(row.ymin):int(row.ymax)]
+
+            if measurement_type == 'max':
+                res.append(subset.max())
+            else:
+                res.append(subset.mean())
+
+        return res
+
+    def _measure_stack(self, measurement_type):
```

Comment:
interestingly, for non-multiplexed experiments (e.g., smFISH) you don't need this measure_stack method right? we should think about how to make the spot detector work well for multiplexed and non-multiplexed methods.

Reply:
Yea, I think some kind of abstraction over hyb rounds. 👍